test/e2e/scale_test.go

```diff
@@ -0,0 +1,65 @@
+package retina
+
+import (
+	"crypto/rand"
+	"math/big"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/microsoft/retina/test/e2e/common"
+	"github.com/microsoft/retina/test/e2e/framework/types"
+	jobs "github.com/microsoft/retina/test/e2e/jobs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestE2ERetina_Scale(t *testing.T) {
+	curuser, err := user.Current()
+	require.NoError(t, err)
+
+	clusterName := curuser.Username + common.NetObsRGtag + strconv.FormatInt(time.Now().Unix(), 10)
+
+	subID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	require.NotEmpty(t, subID)
+
+	location := os.Getenv("AZURE_LOCATION")
+	if location == "" {
+		var nBig *big.Int
+		nBig, err = rand.Int(rand.Reader, big.NewInt(int64(len(locations))))
+		if err != nil {
+			t.Fatalf("Failed to generate a secure random index: %v", err)
+		}
+		location = locations[nBig.Int64()]
+	}
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	// Get to root of the repo by going up two directories
+	rootDir := filepath.Dir(filepath.Dir(cwd))
+
+	chartPath := filepath.Join(rootDir, "deploy", "legacy", "manifests", "controller", "helm", "retina")
+	kubeConfigFilePath := filepath.Join(rootDir, "test", "e2e", "test.pem")
+
+	// CreateTestInfra
+	createTestInfra := types.NewRunner(t, jobs.CreateTestInfra(subID, clusterName, location, kubeConfigFilePath))
+	createTestInfra.Run()
+
+	// Hacky way to ensure that the test infra is deleted even if the test panics
+	defer func() {
+		if r := recover(); r != nil {
+			t.Logf("Recovered in TestE2ERetina_Scale, %v", r)
+			_ = jobs.DeleteTestInfra(subID, clusterName, location).Run()
+		}
+	}()
+
+	// Install Retina
+	installRetina := types.NewRunner(t, jobs.InstallRetina(kubeConfigFilePath, chartPath))
+	types.RunnerWithTimeout(installRetina, 5*time.Minute).Run()
```

Comment:
You should pull out the timeout from `testing.(*T).Deadline` and create a `context.WithDeadline` using that. That puts control back in the user's hands for how long they're willing to wait. They would ultimately control this via the `go test -timeout` flag. It would also respect `<C-c>`

Reply:
Thank you very much for this comment. I was looking for a way to deal with `<C-c>` and `go test -timeout` but had not yet found a good solution. I'll do that.